prov/gni/test/ep.c

```diff
@@ -89,6 +89,39 @@ static void teardown(void)
 
 TestSuite(endpoint, .init = setup, .fini = teardown);
 
+Test(endpoint_info, info)
+{
+	int ret;
+
+	hints = fi_allocinfo();
+	cr_assert(hints, "fi_allocinfo");
+	hints->fabric_attr->name = strdup("gni");
+
+	ret = fi_getinfo(FI_VERSION(1, 0), NULL, 0, 0, hints, &fi);
+	cr_assert(!ret, "fi_getinfo");
+	cr_assert_eq(fi->ep_attr->type, FI_EP_RDM);
+	cr_assert_eq(fi->next->ep_attr->type, FI_EP_DGRAM);
+
+	fi_freeinfo(fi);
+
+	hints->ep_attr->type = FI_EP_RDM;
+	ret = fi_getinfo(FI_VERSION(1, 0), NULL, 0, 0, hints, &fi);
+	cr_assert(!ret, "fi_getinfo");
+	cr_assert_eq(fi->ep_attr->type, FI_EP_RDM);
+	cr_assert_eq(fi->next, NULL);
+
+	fi_freeinfo(fi);
+
+	hints->ep_attr->type = FI_EP_DGRAM;
+	ret = fi_getinfo(FI_VERSION(1, 0), NULL, 0, 0, hints, &fi);
+	cr_assert(!ret, "fi_getinfo");
+	cr_assert_eq(fi->ep_attr->type, FI_EP_DGRAM);
+	cr_assert_eq(fi->next, NULL);
+
+	fi_freeinfo(fi);
+	fi_freeinfo(hints);
+}
```

Comment:
The developer probably added this here since it makes for a simpler quick sanity check.  If we had fi_info as part of a real ```make check```, this sort of code would not be particularly necessary.  Other projects I work on like the sandia-shmem and ucx have nice self-contained checks (in the case of sandia-shmem a real make check) that help a lot in sanity checking a commit(s).

This is kind of related to #1871.  @jsquyres 

I'd be happy to help with adding a functioning make check to libfabric.